### DISCUSSION TITLE
CMI Addons

### DISCUSSION THREAD

**Discussion by hotspuraz** [Zrips/CMI#8496] [Enhancement]
So I've been using CMI for a good 5-6 years now and one thing I've always felt, and I've seen comments by many others, is that CMI is too big. It has a lot of features that most people don't use.

So my suggestion is a big one, more thought provoking than something that is required, as it would take a considerably amount of work.

**The suggestion**
Rather than having one massive plugin with modules you can disable, strip CMI down to an essentials sized base plugin and convert the modules into addons in the format of separate mini-plugins. 
You get a simple easy to manage CMI base plugin that can be used as an Essentials replacement (but better), and you can choose which addons your need to fit your requirements, while not having to load and configure a couple thousand lines of features you don't use. 

Many other plugins already do this successfully. BentoBox comes to mind with their simple base plugin and addons folder. I think it would be a good way to simplify the plugin and remove many features people simply aren't interested in. 
For example: The armorstand editor? Who even uses that and why is it a part of a server's core functionality plugin 😅

**Comment by Zrips** [comment]
CMI - Content Management Interface. Armorstand editor is editor of armor stands where you can fully modify armor stands to your needs, copy over where you want it or create template for either repeated usage by you or creating poses to be sold for regular players. So it is one of those things which can be really useful if you want to utilize armor stands for your server in one or the other way.

Splitting plugin into separate addons would be huge undertaking as entire core of plugin would need to change. We still have module option which gives you more or less same effect.
